src/Microsoft.VisualStudio.Services.Agent/Util/IOUtil.cs

```diff
@@ -333,6 +333,41 @@ public static void CopyDirectory(string source, string target, CancellationToken
             }
         }
 
+        public static void ValidateExecutePermission(string directory)
```

Comment:
I thought about that a bit, but really it is not. The validation check needs the directory to exist. The validation check is so that node/powershell tasks don't fail. Creating the directory could succeed, and the validation check fail. If creating the validation check fails then the customer still get's a good error message.